src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -117,86 +118,118 @@ public bool TryRemoveContext(IAzureContext context)
+                            }
+                            else
+                            {   // if account ID is different from the first tenant account id we need to ignore current tenant
+                                WriteWarningMessage(string.Format(
+                                    ProfileMessages.AccountIdMismatch,
+                                    account.Id,
+                                    tenant,
+                                    accountId));
+                                account.Id = accountId;
+                                token = null;
+                            }
                         }
-                        else if (accountId.Equals(account.Id, StringComparison.OrdinalIgnoreCase))
+                        catch
                         {
-                            account.SetOrAppendProperty(AzureAccount.Property.Tenants, tenant);
-                        }
-                        else
-                        {   // if account ID is different from the first tenant account id we need to ignore current tenant
-                            WriteWarningMessage(string.Format(
-                                ProfileMessages.AccountIdMismatch,
-                                account.Id,
-                                tenant,
-                                accountId));
-                            account.Id = accountId;
-                            token = null;
+                            WriteWarningMessage(string.Format(ProfileMessages.UnableToAqcuireToken, tenant));
                         }
-                    }
-                    catch
-                    {
-                        WriteWarningMessage(string.Format(ProfileMessages.UnableToAqcuireToken, tenant));
-                    }
 
-                    if (token != null &&
-                        newTenant == null &&
-                        TryGetTenantSubscription(token, account, environment, tenant, subscriptionId, subscriptionName, out tempSubscription, out tempTenant))
-                    {
-                        // If no subscription found for the given token/tenant 
-                        // discard tempTenant value unless current token/tenant is the last one.
-                        if (tempSubscription != null || i == (tenants.Count() - 1))
+                        if (token != null &&
+                            newTenant == null &&
+                            TryGetTenantSubscription(token, account, environment, tenant, subscriptionId, subscriptionName, out tempSubscription, out tempTenant))
                         {
-                            newTenant = tempTenant;
-                            newSubscription = tempSubscription;
+                            // If no subscription found for the given token/tenant 
+                            // discard tempTenant value unless current token/tenant is the last one.
+                            if (tempSubscription != null || i == (tenants.Count() - 1))
```

Comment:
Assuming you have a `foreach(var tenant in tenants)` above, this could be

```csharp
tenant == tenants[tenants.Count - 1]
```

Reply:
Also, this `if` can be merged with the enclosing `if` since there are no other statements in the enclosing block.

Unless, is the `out` param from `TryGetTenantSubscription` not available in the `if`-condition?

Reply:
I would agree with whoever actually wrote this that having 5 conditions in the if statement is difficult to read, and that it seems logically to split the two because of the evaluation happening in TryGetTenantSubscription (also the additional comment).